re-edited OptaPlanner release procedure (#2169)
kiegroup/optaplanner: build/release/release-procedure.adoc

=== 1. Release the version in Jira

Navigate to the https://issues.redhat.com/projects/PLANNER?selectedItem=com.atlassian.jira.jira-projects-plugin:release-page[Releases page in JIRA],
find the version to be released and pick “Release” from the Actions. Create a next version for OptaPlanner and release the current version but WITHOUT
release date.

=== 2. Create a new branch for OptaPlanner release

Run on Jenkins https://eng-jenkins-csb-business-automation.apps.ocp-c1.prod.psi.redhat.com/job/KIE/job/optaplanner/job/0-prepare-release-branch[0-prepare-release-branch]
to create a new branch where to release from. +
Most links in this document take the 8.27.x branch as an example!

=== 3. Update Drools

> Since OptaPlanner is not released anymore as a part of the Kogito release pipeline the Drools version is not upgraded automatically. This has to be done by running

Reviewer: To keep it simple, let's focus on the current process without explaining how it used to be.
```suggestion
Run
```

https://eng-jenkins-csb-business-automation.apps.ocp-c1.prod.psi.redhat.com/job/KIE/job/optaplanner/job/8.27.x/job/tools/job/update-drools-optaplanner/[update-drools-optaplanner]. +

> IMPORTANT: this job should be triggered only if the artifacts of Drools version (as an example drools-bom) are already on https://repo1.maven.org/maven2/org/drools/drools-bom[Maven Central].

Reviewer: ```suggestion
IMPORTANT: this job should be triggered only if the artifacts of Drools version (as an example `drools-bom`) are already on https://repo1.maven.org/maven2/org/drools/drools-bom[Maven Central].
```

> Also the Drools version has to be updated on OptaPlanner main branch running

Reviewer: ```suggestion
Also, the Drools version has to be updated on the OptaPlanner main branch by running
```

https://eng-jenkins-csb-business-automation.apps.ocp-c1.prod.psi.redhat.com/job/KIE/job/optaplanner/job/main/job/tools/job/update-drools-optaplanner/[update-drools-optaplanner] on main branch.

=== 4. Release OptaPlanner

> The OptaPlanner release pipeline can be triggered now based on the created new branch in step 2.

Reviewer: ```suggestion
The OptaPlanner release pipeline can now be triggered using the branch created in step 2.
```
